components/message_queue/src/kafka/kafka_impl.rs

```diff
@@ -0,0 +1,396 @@
+            ClientBuilder::new(vec![config.client_config.boost_broker.clone().unwrap()]);
+        if let Some(max_message_size) = config.client_config.max_message_size {
+            client_builder = client_builder.max_message_size(max_message_size);
+        }
+
+        let client = client_builder.build().await.context(Init)?;
+
+        let controller_client = client.controller_client().context(Init)?;
+
+        Ok(Self {
+            config,
+            client,
+            controller_client,
+            topic_client_pool: DashMap::default(),
+        })
+    }
+
+    fn get_or_create_topic_client(
+        &self,
+        topic_name: &str,
+    ) -> std::result::Result<TopicClientRef, RskafkaError> {
+        Ok(self
+            .topic_client_pool
+            .entry(topic_name.to_string())
+            .or_insert(Arc::new(
+                self.client.partition_client(topic_name, PARTITIONS_NUM)?,
+            ))
+            .clone())
+    }
+}
+
+#[async_trait]
+impl MessageQueue for KafkaImpl {
+    type ConsumeAllIterator = KafkaConsumeAllIterator;
+    type Error = Error;
+
+    async fn create_topic_if_not_exist(&self, topic_name: &str) -> Result<()> {
+        // Check in partition_client_pool first, maybe has exist.
+        if self.topic_client_pool.contains_key(topic_name) {
+            info!(
+                "Topic:{} has exist in kafka and connection to the topic is still alive.",
+                topic_name
+            );
+            return Ok(());
+        }
+
+        // Create topic in Kafka.
+        let topic_creation_config = &self.config.topic_creation_config;
+        info!("Try to create topic:{} in kafka", topic_name);
+        let result = self
+            .controller_client
+            .create_topic(
+                topic_name,
+                PARTITIONS_NUM,
+                topic_creation_config.replication_factor,
+                topic_creation_config.timeout_ms,
+            )
+            .await;
+
+        match result {
+            // Race condition between check and creation action, that's OK.
+            Ok(_) | Err(RskafkaError::ServerError(ProtocolError::TopicAlreadyExists, ..)) => Ok(()),
+
+            Err(e) => Err(e).context(CreateTopic {
+                topic_name: topic_name.to_string(),
+            }),
+        }
+    }
+
+    async fn produce(&self, topic_name: &str, message: Vec<Message>) -> Result<Vec<i64>> {
+        let topic_client = self
+            .get_or_create_topic_client(topic_name)
+            .context(Produce {
+                topic_name: topic_name.to_string(),
+            })?;
+
+        let records: Vec<Record> = message.into_iter().map(|m| m.into()).collect();
+        Ok(topic_client
+            .produce(records, Compression::default())
+            .await
+            .context(Produce {
+                topic_name: topic_name.to_string(),
+            })?)
+    }
+
+    async fn consume_all(&self, topic_name: &str) -> Result<KafkaConsumeAllIterator> {
```

Comment:
Why not just call it consume? And call `ConsumeAllIterator` as `ConsumeIterator`?

Reply:
Because in message queue, `consume` usually means consume one by one and should return one message per time, but the method here will return all the data in the topic by a iterator.
I propose to  implement the `consume` method in the future for the logic of fetching process from master in the slave.

Renaming the iterator to `ConsumeIterator ` is reasonable, I will do it.